applications/coreutils/wc.cpp

```diff
@@ -0,0 +1,169 @@
+#include <libsystem/Result.h>
+#include <libutils/ArgParse.h>
+#include <libsystem/io/Stream.h>
+
+struct FileInfo {
+  char* filename;
+  int bytes;
+  int chars;
+  int lines;
+  int words;
+  Result res
+};
+
+/*Finding Number of new line characters in a buffer*/
+int line_count(const char * str) {
+  int liness = 0, i = 0;
+  for (i = 0; str[i] != '\0'; i++) {
+    if (str[i] == '\n') {
+      liness += 1;
+    }
+  }
+  return lines;
+}
+
+int word_count(const char * str) {
+  int words = 0, i = 0;
+  // skip initial white space characters
+  while (str[i] != '\0') {
+    if (str[i] == ' ' || str[i] == '\n' || str[i] == '\r' || str[i] == '\t') {
+      i++;
+    } else {
+      break;
+    }
+  }
+  for (; str[i] != '\0'; i++) {
+    if (str[i] == ' ' || str[i] == '\n' || str[i] == '\r' || str[i] == '\t') {
+      words += 1;
+      // skip consucative white space
+      while (str[i] == ' ' || str[i] == '\n' || str[i] == '\r' || str[i] == '\t') {
+        i++;
+      }
+    }
+  }
+  return words;
+}
+
+void printFileInfo(bool* params, bool all, FileInfo* fl) {
+  if(lines || all) {
+    stream_format(out_stream, "%d ", fl->lines);
+  }
+  if(bytes || all) {
+    stream_format(out_stream, "%d ", fl->bytes);
+  }
+  if(words || all) {
+    stream_format(out_stream, "%d ", fl->words);
+  }
+  if(characters || all) {
+    stream_format(out_stream, "%d ", fl->chars);
+  }
+
+  stream_format(out_stream, "%s\n", fl->filename)
+  
+}
+
+FileInfo wc(char* path) {
+  //Initialization 
+  FileInfo to_return;
+  to_return.filename = path;
+
+  __cleanup(stream_cleanup) Stream *stream = stream_open(path, OPEN_READ);
+  if (handle_has_error(stream)) {
+    process_exit(PROCESS_ERROR);
+  }
+
+  // Get number of bytes from file
+
+  FileState stat = {};
+  stream_stat(stream, &stat);
+  to_return.bytes = stat.size;
```

Comment:
This won't work if you pipe into wc

Reply:
Do you have any other ideas on how I could get the size of the file in bytes then?